hooks/environment

```diff
@@ -12,6 +12,22 @@ check_command() {
   fi
 }
 
+# the buildkite agent redactor has been added from version 3.66.0 and on. So we need to check it.
+# https://github.com/buildkite/agent/releases/tag/v3.66.0
+check_buildkite_version() {
+  local version_output
+  version_output=$(buildkite-agent --version)
+  local version
+  version=$(echo "$version_output" | grep -oP '\d+\.\d+\.\d+')
+  local min_version="3.66.0"
+  if [ "$(printf '%s\n' "$min_version" "$version" | sort -V | head -n1)" = "$min_version" ]; then
```

Comment:
This only prints out one version - was that expected?
```
printf '%s\n' "$min_version" "$version"
```

Reply:
No, this does not print out one version. It prints out the min version and the version each in a different line, so they can be sorted afterwards.

You can try this in your bash:
```
printf '%s\n' "nikos" "jamie"
```

it will print out:
```
nikos
jamie
```

Reply:
Ah sorry - my bad.

That being said, I think this is more because `printf` will 

So we should correct our format string to make this super explicit, or replace the usage of `printf`

```suggestion
  if [ "$(printf '%s\n%s' "$min_version" "$version" | sort -V | head -n1)" = "$min_version" ]; then
```

or


```suggestion
  if [ "$(echo -e "$min_version\n$version" | sort -V | head -n1)" = "$min_version" ]; then
```

Reply:
The current set is working fine. I cannot understand what cases do you refer to. Does your proposal add somethng extra or is it a matter of taste?

Reply:
It works, yes, but is confusing - I would expect the format string to contain _all_ values being printed out, so it's clear what the format of the output is

Reply:
ok If you find it confusing lets fix it

Reply:
> however this is a general idea of how printf works

I think yes - but it can be confusing i.e. coming from Go's `fmt.Printf`, where additional arguments are _not_ output

Reply:
If we compare bash style to go -woohoo- we have a lot to discuss :D anyway, I will fix it.

Reply:
changed in 8341afb07604de424ad9bd1e0880dc705022ef94